games-action/another_world/another_world-0.0.1.recipe

```diff
@@ -0,0 +1,53 @@
+SUMMARY="Another World (NEO-RAW)"
+DESCRIPTION="RAW is a re-implementation of the engine used \
+in the game Another World. This game, released \
+under the name Out Of This World in non-European \
+countries, was written by Eric Chahi at the beginning of the '90s. \
+YOU NEED ORIGINAL GAME FILES (DOS VERSION)."
+HOMEPAGE="https://github.com/fabiensanglard/Another-World-Bytecode-Interpreter"
+COPYRIGHT="2011-2020 Fabien Sanglard"
+LICENSE="GNU GPL v2"
+REVISION="1"
+SOURCE_URI="https://github.com/fabiensanglard/Another-World-Bytecode-Interpreter/archive/dea6914a82f493cb329188bcffa46b9d0b234ea6.zip"
+CHECKSUM_SHA256="f75d33acc822de4a33bce5fc8781839c8e66f2a62720a1290f3968f8c12e0439"
+SOURCE_DIR="Another-World-Bytecode-Interpreter-dea6914a82f493cb329188bcffa46b9d0b234ea6"
+ADDITIONAL_FILES="another_world.sh"
+
+ARCHITECTURES="all !x86_gcc2"
+SECONDARY_ARCHITECTURES="x86"
+
+PROVIDES="
+	another_world$secondaryArchSuffix = $portVersion
+	app:AnotherWorld = $portVersion
+	"
+REQUIRES="
+	haiku$secondaryArchSuffix
+	lib:libSDL2_2.0$secondaryArchSuffix
+	lib:libz$secondaryArchSuffix
+	"
+
+BUILD_REQUIRES="
+	haiku${secondaryArchSuffix}_devel
+	devel:libSDL2_2.0$secondaryArchSuffix
+	devel:libz$secondaryArchSuffix
+	"
+BUILD_PREREQUIRES="
+	cmd:cmake
+	cmd:gcc$secondaryArchSuffix
+	cmd:make
+	"
+
+BUILD()
+{
+	cmake -DCMAKE_BUILD_TYPE=Release .
+	make
+}
+
+INSTALL()
+{
+	mkdir -p $appsDir/AnotherWorld/
+	cp raw $appsDir/AnotherWorld/raw
```

Comment:
For me I would just put the binary in $binDir (or $prefix/bin), that way the user can always control the binary itself even without using the script.

Reply:
Should I create "AnotherWorld" directory in $binDir? I ask because:
1) "raw" is a rather common name, it may conflict with other binaries
2) should "another_world.sh" (the launch script) be put there too?

Reply:
Got a point here, same goes however in $appsDir, renaming it would be weird too, so no strong opinion atm, maybe some others can chime in here?

Reply:
Oh, and no subdirectories in $binDir please :)

Reply:
Question is, as a game, will it often be started from the commandline? Will it benefit from being in the PATH in /bin?
If not, putting it in $appsDir feels more correct.

Reply:
@Begasus and @humdingerb, my opinion is that, since you always have to specify at least the --datapath parameter, not many people will benefit from launching the "raw" executable directly. At that point, I think most users will write their own launch script.

But this is just my opinion, if having the bin in $binDir is something you want, I'm ok to move it there ;)

Reply:
Maybe this question should be posed to a wider audience, but my feeling is that the /bin folder is where the CLI apps live. Having a GUI app or game there feels strange to me, but that may be personal preferece or unfamiliarity with how other OS handle it.

Reply:
Other OS don't make that distinction.

Linux has only "bin" and Windows has only "Program Files" (which is more like apps/)

As often, we go with the "why not both?" approach.

I think a subdirectory in appsDir makes sense here.

Reply:
@pulkomandy Linux also has the "opt" directory, which is more similar to "Program Files" in Windows, and I think it is like "appsDir" in Haiku. That's why I think it is ok to use a subdirectory there, where to place both the binary and the startup script. But again, this is just personal preference :) 

Another solution that can make everyone happy: 
1) rename the game binary from "raw" to "neo-raw" (to avoid any name clashes)
2) copy the new "neo-raw" binary to $binDir
3) create the "AnotherWorld" directory in $appDir, then copy the startup script and symlink "neo-raw" here

This way the user has the neo-raw binary in $binDir and can launch it from terminal if he wants, and at the same time the game is also bootable from the startup script.

Reply:
Or go with your 1) and 2), but put the startup script as "Another World" directly in $appDir. No need for a sub-folder with just the one file.

Reply:
Agreed with @humdingerb  :)

Reply:
Perfect, I agree with @humdingerb too. I'll patch the script later tonight :)